samples/static_library/CMakeLists.txt

```diff
@@ -79,17 +79,19 @@ DEPS
   simple_mul
 )
 
-iree_lit_test(
-  NAME
-    static_library_demo_test
-  TEST_FILE
-    "static_library_demo_test.txt"
-  TOOLS
-    ::static_library_demo
-    FileCheck
-  LABELS
-    "hostonly"
-)
+if(IREE_BUILD_TESTS)
```

Comment:
iree_lit_test should disable itself ala iree_cc_test - iree_lit_test_suite does, but no clue why iree_lit_test doesn't (@ScottTodd was that intentional?) - I'd rather us make those rules consistent instead of doing this

Reply:
Yeah `iree_lit_test` is missing this code: https://github.com/iree-org/iree/blob/63a2d141adf7b3911bc94c7f779e1896fb25aecb/build_tools/cmake/iree_lit_test.cmake#L100-L103

Just an oversight - I only added a direct use of `iree_lit_test` recently. We had been exclusively using it through the wrapper `iree_lit_test_suite` before.